File: python/169-Majority Element.py

def majorityElement( nums) :
    numsDict = {}
    for i in range(len(nums)):
        if nums[i] not in numsDict.keys():
            numsDict[nums[i]] = 1
        else:
            numsDict[nums[i]] +=1

    for k, v in numsDict.items():
        if (v>(int(len(nums)/2))):
            return k
            

print(majorityElement(nums = [2,2,1,1,1,2,2]))

File: javascript/array + string/169-majorityElement.js
var majorityElement = function (nums) {
  let n = nums.length;
  let maxCount = 0;
  let item = 0;
  nums.sort((a, b) => a - b);
  let countMap = new Map();
  for (let item of nums) {
    countMap.set(item, (countMap.get(item) || 0) + 1);
  }
  console.log(countMap);

  for (let [key, value] of countMap) {
    if (value > maxCount) {
      maxCount = value;
      item = key;
    }
  }

  return item;
};

let nums = [2, 2, 1, 1, 1, 2, 2];
majorityElement(nums);
